BWAPI4J/src/main/java/org/openbw/bwapi4j/UnitStatCalculator.java

```diff
@@ -125,6 +127,14 @@ public int weaponDamageCooldown(UnitType unitType) {
         return cooldown;
     }
 
+    /**
+     * Retrieves the weapon cooldown of a unit.
+     */
+    public int weaponDamageCooldown(Unit unit) {
+
+        return ((GroundAttacker)unit).getGroundWeapon().cooldown();
```

Comment:
There's no way of knowing if this cast will succeed until runtime. Therefore, the `Unit` parameter type should be `GroundAttacker` already and the method can be called internally via `weaponDamageCooldown(this)` from `GroundAttacker` units. I will change this in an incoming PR.